source/common/quic/envoy_quic_client_stream.cc

```diff
@@ -57,7 +57,11 @@ Http::Status EnvoyQuicClientStream::encodeHeaders(const Http::RequestHeaderMap&
       sent_head_request_ = true;
     }
   }
+  uint64_t bytes_written = stream_bytes_written();
   WriteHeaders(std::move(spdy_headers), end_stream, nullptr);
+  mutableBytesMeter()->addHeaderBytesSent(stream_bytes_written() - bytes_written);
+  mutableBytesMeter()->addWireBytesSent(stream_bytes_written() - bytes_written);
```

Comment:
Can these be wrapped into some scoped object doing the bytes calculation?

Reply:
Hm. That's a great idea!